internal/driver/internal/k8s/logicalvolume_service.go

```diff
@@ -313,36 +312,39 @@ func (s *LogicalVolumeService) ExpandVolume(ctx context.Context, volumeID string
 		return err
 	}
 
-	// wait until topolvm-node expands the target volume
-	for {
-		logger.Info("waiting for update of 'status.currentSize'", "name", lv.Name)
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		case <-time.After(1 * time.Second):
-		}
-
+	return wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Duration: 1 * time.Second, // initial backoff
+		Factor:   1,               // factor for duration increase
+		Jitter:   0.1,
+		Cap:      300 * time.Second, // wait up to 10 seconds per retry, 1, 2, 4, 8, 10 seconds.
+		Steps:    300,               // run at most 10 times
+	}, func(ctx context.Context) (done bool, err error) {
 		var changedLV topolvmv1.LogicalVolume
-		err := s.getter.Get(ctx, client.ObjectKey{Name: lv.Name}, &changedLV)
-		if err != nil {
+		if err := s.getter.Get(ctx, client.ObjectKey{Name: lv.Name}, &changedLV); err != nil {
 			logger.Error(err, "failed to get LogicalVolume", "name", lv.Name)
-			return err
+			return true, err
 		}
+
 		if changedLV.Status.Code != codes.OK {
-			return status.Error(changedLV.Status.Code, changedLV.Status.Message)
+			return true, status.Error(changedLV.Status.Code, changedLV.Status.Message)
```

Comment:
```suggestion
			return false, status.Error(changedLV.Status.Code, changedLV.Status.Message)
```

Reply:
See my message above, is retrying this instead of failing fast desired?

Reply:
not desired, but I assume you know, the value is not meaningless if an error returns, so I prefer the zero value.